src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,402 @@
+package duke;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileNotFoundException;
+
+public class Duke {
+
+    public static final Path dataDirectory = Path.of(System.getProperty("user.dir") + "\\data");
+
+    public static final Path dataFileDirectory = Path.of(System.getProperty("user.dir") + "\\data" + "\\duke.txt");
+
+    public static final String LINE = "____________________________________________________________";
+
+    public static void writeToDisk(ArrayList<Task> tasks) throws IOException{
+        //Each time I want to write to the Disk, I clear away the entire contents of the data file and write everything
+        //Done by deleting the file and creating it again
+        File clear = new File(String.valueOf(dataFileDirectory));
+        clear.delete();
+        createNewFile();
+
+        //To write to the new file, I will append 1 task from the ArrayList at a time
+        FileWriter fw = new FileWriter(String.valueOf(dataFileDirectory), true);
+        for(Task task : tasks){
+            String temp = "";
+
+            if(task.getDoneStatus()){
+                temp += "1|";
+            } else {
+                temp += "0|";
+            }
+            temp += task.getType() + "|" + task.getDescription() + "|" + task.getSeparator() + "|";
+            if(task instanceof Event) {
+                temp += ((Event) task).getAt();
+            } else if(task instanceof Deadline){
+                temp += ((Deadline) task).getBy();
+            } else {
+                temp += "";
+            }
+            temp += "\n";
+            fw.write(temp);
+        }
+        fw.close();
+    }
+
+    public static void wrapWriteToDisk(ArrayList<Task> tasks){
+        try{
+            writeToDisk(tasks);
+        } catch (IOException e) {
+            System.out.println("Issues with writing to .txt file");
```

Comment:
The message prints there is an error occurs when writing files, but what error exactly? This problem happens several times in later scripts (the exception catch does not show what error actually occurred). Maybe you can print out the content of the exception variable "e".